### PR TITLE
all tests have passed

### PR DESCRIPTION
**Pull request by VChorney**

### CODE REVIEW

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 1]
```diff
@@ -51,10 +61,71 @@ public boolean retainAll(Collection<?> c) {
     }
 
     public boolean removeAll(Collection<?> c) {
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java]: Rename variable used in this method to have some sense.

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 2]
```diff
@@ -51,10 +61,71 @@ public boolean retainAll(Collection<?> c) {
...
+        Iterator var3;
+        if (this.size() > c.size()) {
+            for(var3 = c.iterator(); var3.hasNext(); var2 |= this.remove(var3.next())) {
+            }
+        } else {
+            var3 = this.iterator();
+
+            while(var3.hasNext()) {
+                if (c.contains(var3.next())) {
+                    var3.remove();
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java]: why do you remove from collection? You have to remove from map.
Review method logic.

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 3]
```diff
@@ -51,10 +61,71 @@ public boolean retainAll(Collection<?> c) {
...
 
     }
+    public static <E> Range<E> of(E first, E second) {
+        if (first instanceof Integer) {
+            return rangeInt((Integer) first, (Integer) second);
+        }
+        return rangeFloat((Float) first, (Float) second);
+    }
+
+    public static <E> Range<E> of(Character first, Character second, Function function) {
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java]: You have to create an general solution for of method

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 4]
```diff
@@ -51,10 +61,71 @@ public boolean retainAll(Collection<?> c) {
...
+            }
+        }
+
+        return var2;
     }
 
     public void clear() {
 
     }
+    public static <E> Range<E> of(E first, E second) {
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java]: Use overloading for such logic.

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 3 above)
Comment (VChorney) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java]: > You have to create an general solution for of method
What do you mean?
Like generic or overloading or what?

**Pull request closed without merging**